application/backend/app/api/routers/dataset_revisions.py

```diff
@@ -33,33 +40,74 @@
     },
 )
 def list_dataset_revision_items(
-    _project: Annotated[Project, Depends(get_project)],
-    _dataset_service: Annotated[DatasetService, Depends(get_dataset_service)],
-    _dataset_revision: Annotated[DatasetRevision, Depends(get_dataset_revision)],
-    _limit: Annotated[int, Query(ge=1, le=MAX_DATASET_ITEMS_NUMBER_RETURNED)] = DEFAULT_DATASET_ITEMS_NUMBER_RETURNED,
-    _offset: Annotated[int, Query(ge=0)] = 0,
-    _subset: Annotated[DatasetItemSubset | None, Query()] = None,
-) -> DatasetItemsWithPagination:
+    project: Annotated[Project, Depends(get_project)],
+    dataset_revision_service: Annotated[DatasetRevisionService, Depends(get_dataset_revision_service)],
+    dataset_revision: Annotated[DatasetRevision, Depends(get_dataset_revision)],
+    limit: Annotated[int, Query(ge=1, le=MAX_DATASET_ITEMS_NUMBER_RETURNED)] = DEFAULT_DATASET_ITEMS_NUMBER_RETURNED,
+    offset: Annotated[int, Query(ge=0)] = 0,
+    subset: Annotated[DatasetItemSubset | None, Query()] = None,
+) -> DatasetItemsRevisionWithPagination:
     """List the items in a dataset revision. This endpoint supports pagination."""
-    raise NotImplementedError
+    items_data, total_count = dataset_revision_service.list_dataset_revision_items(
+        project_id=project.id,
+        dataset_revision=dataset_revision,
+        limit=limit,
+        offset=offset,
+        subset=subset,
+    )
+
+    items = []
+    for item_data in items_data:
+        item_view = DatasetItemRevisionView(
+            id=UUID(item_data["id"]),
+            name=Path(item_data["image"]).stem,
+            format=DatasetItemFormat(Path(item_data["image"]).suffix.lstrip(".")),
+            width=item_data["image_info"]["width"],
+            height=item_data["image_info"]["height"],
+            subset=DatasetItemSubset(item_data["subset"]),
+        )
+        items.append(item_view)
```

Comment:
Ideally, this logic should be in `DatasetRevisionService.list_dataset_revision_items`, which should return an object of type `app.models.dataset_item_revision.DatasetRevisionItem` (the domain entity) instead of a raw dictionary. The endpoint should then convert to the view type with `DatasetItemRevisionView.model_validate(dataset_revision_item, from_attributes=True)`.

For reference, you can check the implementation of `list_dataset_items` in `api/routers/datasets.py`.